content/docs/guides/opentelemetry.md

```diff
@@ -0,0 +1,208 @@
+    return jsonify({"result": result})
+
+@app.route('/multiply', methods=['POST'])
+def multiply():
+    data = request.json
+    result = data['num1'] * data['num2']
+    return jsonify({"result": result})
+
+@app.route('/divide', methods=['POST'])
+def divide():
+    data = request.json
+    if data['num2'] == 0:
+        return jsonify({"error": "Division by zero is not allowed."}), 400
+    result = data['num1'] / data['num2']
+    return jsonify({"result": result})
+
+if __name__ == '__main__':
+    app.run(debug=True)
+
+```
+
+Start the application and send a couple of requests:
+
+``` shell
+$ python app.py
+ * Serving Flask app 'app'
+ * Debug mode: on
+WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
+ * Running on http://127.0.0.1:5000
+Press CTRL+C to quit
+ * Restarting with stat
+ * Debugger is active!
+ * Debugger PIN: 988-101-184
+127.0.0.1 - - [03/Sep/2023 09:48:09] "POST /add HTTP/1.1" 200 -
+```
+
+``` shell
+$ curl -X POST -H "Content-Type: application/json" -d '{"num1": 5, "num2": 100}' http://127.0.0.1:5000/add
+
+{
+  "result": 105
+}
+```
+
+## Auto Instrumentation
+
+Let's add the OpenTelemetry instrumentation now. We will use Otel's Flask instrumentation library which auto-instruments all request handelers.
+
+``` shell
+pip opentelemetry-api opentelemetry-sdk opentelemetry-exporter-otlp opentelemetry-instrumentation-flask
+```
+
+Then update the code as follows:
+
+``` shell
+from flask import Flask, jsonify, request
+from opentelemetry import metrics
+from opentelemetry.exporter.otlp.proto.http.metric_exporter import OTLPMetricExporter
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.instrumentation.flask import FlaskInstrumentor
+
+# Configure the OTLP metrics exporter
+otlp_exporter = OTLPMetricExporter(endpoint="http://localhost:9090/api/v1/otlp/v1/metrics")
+reader = PeriodicExportingMetricReader(otlp_exporter, 1000)
+resource = Resource.create({"service.name": "calculator"})
+provider = MeterProvider(metric_readers=[reader], resource=resource)
+
+metrics.set_meter_provider(provider)
+meter = metrics.get_meter(__name__)
+
+app = Flask(__name__)
+FlaskInstrumentor().instrument_app(app)
+```
+
+A lot is happening is here, let's break it down.
+
+- We created the OTLP Exporter with the Prometheus endpoint which accepts the OpenTelemetry metrics in native format. The endpoint path is `/api/v1/otlp/v1/metrics`.
+- We setup required `reader`, `resource` and `provider` for metrics instrumentation.
+- We wrapped the Flask app in `FlaskInstrumentor's` `instrument_app` method so that the auto instrumentation will kick in.
+
+When the application is run and few requests are sent to it, the auto instrumented metrics will be **pushed** from the application to Prometheus via the `FlaskInstrumentor`.
```

Comment:
This can be misunderstood as its the FlaskInstrumentor that pushes the metrics. If i understand correctly, what happens is that FlaskInstrumentor only *collects* the metrics internally. PeriodicExportingMetricReader+OTLPMetricExporter does the push.